### PR TITLE
Convert wormhole payload to borsh

### PR DESCRIPTION
**Pull request by kiseln**

### CODE REVIEW

**Comment by karim-en** [comment]
@kiseln could you please fix the conflicts

File: near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs [hunk 1]
```diff
@@ -111,49 +111,50 @@ impl ParsedVAA {
...
 
-    struct FinTransferWh {
-        string token;
-        uint128 amount;
-        string recipient;
-        uint128 nonce;
-    }
+#[derive(Debug, BorshDeserialize)]
+struct FinTransferWh {
+    _token: String,
```
Comment (karim-en) [inline near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs:122]: lets add the type (first byte) to this struct and move the verification to the try_into.
see this https://github.com/Near-One/omni-bridge/pull/106#discussion_r1834088215

Comment (kiseln) [reply near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs:122]: I still like verifying inside `verify_vaa_callback`. We can forget and remove the check inside the conversion methods and introduce a vulnerability this way. However, I don't mind making this check in both places

Comment (kiseln) [reply near/omni-prover/wormhole-omni-prover-proxy/src/parsed_vaa.rs:124]: 50f55ccb6f04ec9987a25384a537516cda4dd51a

**Comment by karim-en** [comment]
@kiseln please apply the fmt

**Review by karim-en — APPROVED** [review]

**Review by karim-en — APPROVED** [review]

**Pull request merged**